test/integration/scheduler/util.go

```diff
@@ -381,7 +381,8 @@ func nodeTainted(cs clientset.Interface, nodeName string, taints []v1.Taint) wai
 			return false, err
 		}
 
-		if len(taints) != len(node.Spec.Taints) {
```

Comment:
In terms of the testcase, if both TaintBasedEviction and TaintNodeByCondtions are enabled, there should be at least 2 taints (`NotReady:NoExecute` and `NotReady:NoSchedule`) applied, so comparing the size is absolutely incorrect.

And also the comment said "contain", the old logic doesn't follow either.